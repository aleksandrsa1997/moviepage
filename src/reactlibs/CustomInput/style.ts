import {createStyles, makeStyles, Theme} from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            margin: theme.spacing(1),
        },
        styleInput: {
            height: '100%',
            width: '100%',
            border: 'none'
        }
    }),
);