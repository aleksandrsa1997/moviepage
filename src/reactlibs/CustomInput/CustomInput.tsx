import React from 'react';
import Input from '@material-ui/core/Input';
import { useStyles } from './style';
import { useTranslation } from 'react-i18next';

type TProp = {
    placeholder?: string;
    onChange?: (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
    value?: string;
}

const CustomInput = ({ placeholder = 'enter_text', onChange, value }: TProp) => {
    const { t } = useTranslation();
    const classes = useStyles();

    return (
        <Input
            value={value}
            onChange={onChange}
            className={classes.styleInput}
            placeholder={t(placeholder)}
        />
    );
}

export  default CustomInput;