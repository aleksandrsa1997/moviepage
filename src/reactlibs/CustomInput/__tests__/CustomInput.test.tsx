import React from 'react';
import CustomInput from '../CustomInput';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('CustomInput tests', () => {
    let props;
    let component;

    beforeEach(()=> {
        props = {
            placeholder: 'enter text ...',
            value: 'test',
            onChange: jest.fn()
        }
        component = mount(<CustomInput {...props} />);
    })

    it('should render snapshot', () => {

        expect(component).toMatchSnapshot();
    });

    it('should call onChange function', () => {
        act(() => {
            component.find('CustomInput').at(0).getElement().props.onChange();
        })
        
        component.update();
        expect(props.onChange).toHaveBeenCalled();
    })
});