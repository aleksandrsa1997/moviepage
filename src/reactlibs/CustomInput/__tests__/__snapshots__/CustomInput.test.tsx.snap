// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput tests should render snapshot 1`] = `ReactWrapper {}`;
